goal_src/jak3/engine/collide/collide-cache.gc

```diff
@@ -7,3 +7,898 @@
+      (set! (-> a1-5 ccache) this)
+      (set! (-> a1-5 prim-core world-sphere quad) (-> v1-34 prim-core world-sphere quad))
+      (set! (-> a1-5 prim-core quad 1) (-> v1-34 prim-core quad 1))
+      )
+    (+! (-> this num-prims) 1)
+    (+! (-> this num-tris) 2)
+    0
+    (none)
+    )
+  )
+
+(defmethod fill-using-bounding-box ((this collide-cache) (arg0 collide-query))
+  (rlet ((vf0 :class vf)
+         (vf1 :class vf)
+         (vf2 :class vf)
+         (vf3 :class vf)
+         (vf4 :class vf)
+         )
+    (init-vf0-vector)
+    (+! (-> *collide-stats* calls) 1)
+    (new 'static 'vector :x 0.5)
+    (nop!)
+    (let ((v1-4 (-> arg0 collide-with-s32)))
+      (nop!)
+      (let ((a0-2 (-> arg0 ignore-process0)))
+        (nop!)
+        (let ((a1-1 (-> arg0 ignore-process1)))
+          (nop!)
+          (let ((a2-0 (-> arg0 ignore-pat-s32)))
+            (nop!)
+            (.lvf vf1 (&-> arg0 bbox min quad))
+            (nop!)
+            (.lvf vf2 (&-> arg0 bbox max quad))
+            (nop!)
+            (set! (-> this ignore-processes 0) (the-as process a0-2))
+            (nop!)
+            (set! (-> this ignore-processes 1) (the-as process a1-1))
+            (.mov.vf vf1 vf0 :mask #b1000)
+            (nop!)
+            (.mov.vf vf2 vf0 :mask #b1000)
+            (set! (-> this ignore-mask) (the-as pat-surface a2-0))
+            )
+          )
+        )
+      (.ftoi.vf vf3 vf1)
+      (nop!)
+      (.ftoi.vf vf4 vf2)
+      (set! (-> this num-tris) 0)
+      (nop!)
+      (.svf (&-> this collide-box min quad) vf1)
+      (nop!)
+      (.svf (&-> this collide-box max quad) vf2)
+      (nop!)
+      (.svf (&-> this collide-box4w min quad) vf3)
+      (nop!)
+      (.svf (&-> this collide-box4w max quad) vf4)
+      (nop!)
+      (.svf (&-> arg0 bbox4w min quad) vf3)
+      (nop!)
+      (.svf (&-> arg0 bbox4w max quad) vf4)
+      (nop!)
+      (set! (-> this num-prims) 0)
+      (nop!)
+      (set! (-> this collide-with) (the-as collide-spec v1-4))
+      )
+    0
+    (when (logtest? (-> arg0 collide-with) (collide-spec backgnd))
+      (let* ((a0-3 this)
+             (t9-0 (method-of-object a0-3 fill-from-bg))
+             (a1-2 (method-of-type collide-hash drawable-method-11))
+             (a2-1 collide-list-fill-bg-using-box)
+             (a3-0 arg0)
+             )
+        (t9-0 a0-3 (the-as (function collide-hash int collide-list collide-query int) a1-2) a2-1 a3-0)
+        (dotimes (s4-0 (-> *level* length))
+          (let ((v1-15 (-> *level* level s4-0)))
+            (when (= (-> v1-15 status) 'active)
+              (let ((a0-6 (-> v1-15 bsp hfrag-drawable)))
+                (if (nonzero? a0-6)
+                    ((-> (the-as process-drawable (-> a0-6 type)) event-hook)
+                     (the-as process a0-6)
+                     (the-as int this)
+                     (the-as symbol arg0)
+                     (the-as event-message-block a3-0)
+                     )
```

Comment:
I think this is a bad cast here - the `hfrag-drawable` in `bsp-header` is probably something else. But probably fine to leave for now if we don't know what it is.

Reply:
Whoops, yeah I checked the type in some bsp and it's an `hfragment`, I did use this cast somewhere else later in the file, but forgot to fix this here.